.circleci/config.yml

```diff
@@ -0,0 +1,279 @@
+    steps:
+    - run:
+        name: Configuring PATH
+        command: |
+          echo >> ${BASH_ENV} 'export PATH=~/bin:~/repo/venv/bin:${PATH}'
+          echo >> ${BASH_ENV} '. ~/.nvm/nvm.sh'
+jobs:
+  run-flake8:
+    executor: ubuntu-builder
+
+    steps:
+      - checkout
+      - config-path
+      - restore_cache:
+          key: v1-flake8-venv-{{ checksum "constraints.txt" }}
+      - create-venv
+      - run:
+          name: Install flake8
+          command: |
+            pip install flake8 flake8-string-format flake8-per-file-ignores -c constraints.txt
+      - save_cache:
+          key: v1-flake8-venv-{{ checksum "constraints.txt" }}
+          paths:
+            - venv
+      - run:
+          name: Run flake8
+          command: |
+            flake8 py-bin py-deploy tests
+  run-solium:
+    executor: ubuntu-builder
+
+    steps:
+      - checkout
+      - config-path
+      - run:
+          name: Install solium
+          command: |
+            npm install -g 'solium@>=1.0.9'
+      - run:
+          name: Run solium
+          command: |
+            solium --dir contracts
+  install:
+    executor: ubuntu-builder
+    steps:
+      - checkout
+      - config-path
+      - create-venv
+
+      # install populus and compile the contracts
+      - restore_cache:
+          key: populus-venv-{{ checksum "constraints-populus.txt" }}
+      - run:
+          name: Install populus
+          command: |
+            make venv-populus
+      - save_cache:
+          key: populus-venv-{{ checksum "constraints-populus.txt" }}
+          paths:
+            - venv-populus
+      - run:
+          name: Compile contracts
+          command: |
+            make compile
+
+      # install py-bin/py-deploy
+
+      - restore_cache:
+          key: venv-{{ checksum "constraints.txt" }}-{{ checksum "requirements.txt" }}
+      - run:
+          name: Install requirements
+          command: |
+            make install-requirements
+      - save_cache:
+          key: venv-{{ checksum "constraints.txt" }}-{{ checksum "requirements.txt" }}
+          paths:
+            - venv
+            - .requirements-installed
+      - run:
+          name: Install py-bin/py-deploy
+          command: |
+            make install
+      - run:
+          name: Build python distributions
+          command: |
+            make dist
+      - persist_to_workspace:
+          root: "~"
+          paths:
+            - repo
+
+  upload-pypi:
+    executor: ubuntu-builder
+    steps:
+      - attach_workspace:
+          at: '~'
+      - config-path
+      - run:
+          name: Init .pypirc
+          command: |
+            echo -e "[pypi]" >> ~/.pypirc
+            echo -e "username = $PYPI_USER" >> ~/.pypirc
+            echo -e "password = $PYPI_PASSWORD" >> ~/.pypirc
+      - run:
+          name: Upload to pypi
+          command: |
+            twine upload py-bin/dist/* py-deploy/dist/*
+
+  upload-npm:
```

Comment:
In the travis CI we have to npm uploads. The one with the dev tag for every push to develop, and the one with the latest tag for every push with a version tag:

https://github.com/trustlines-network/contracts/blob/develop/.travis.yml#L114